astropy/time/formats.py

```diff
@@ -72,6 +73,11 @@ def _regexify_subfmts(subfmts):
     return tuple(new_subfmts)
 
 
+class TimeMaskedArray(np.ma.MaskedArray):
```

Comment:
I don't like the `masked_array` repr because:
- I don't need to see an explicit `data = ..`, which has `--` marking masked elements, AND `mask = [..]`, which is basically the same information.
- The `fill_value` is just a silly thing to include.  It is irrelevant *most* of the time.
- The `data = ` is redundant.  I know it is the data.

One can say this uses an old-school definition of repr, that it gives you the info to reproduce the object (although there it fails because it doesn't include dtype). But I think in this age of Jupyter notebook / interactive analysis, it is more important to have the object repr be a concise and visually-friendly representation of the object.